### ISSUE TITLE
When an array has more than 21 items, it is parsed from a url string into an object, instead of array

### ISSUE THREAD

**Issue by carpben** [ljharb/qs#324]
Following an error that array.includes is not a function, it turns out that when an array which is longer than 21 items is translated into a url query string (using qs.stringify), and than the url string is parsed (using qs.parse) it returns an object, not an array. This can obviously cause issues when you expect an array and it's methods. 

```js
console.log( typeof(qs.parse(qs.stringify({activities:[....array 20 items long]})) ) // Array
console.log( typeof(qs.parse(qs.stringify({activities:[....array 22 items long]})) ) // Object
```

**Comment by ljharb** [comment]
This is the `arrayLimit` option, which is documented [in the readme](https://github.com/ljharb/qs#parsing-arrays).

**Comment by carpben** [comment]
@ljharb Thanks

**Comment by samicodesit** [comment]
This is happening to me but with `qs.stringify`, not with `parse`
and I can see there's no `arrayLimit` option for the `stringify` method.

Any ideas?

**Comment by ljharb** [comment]
@samicodesit can you provide the exact code that you used with stringify? Filing it as a new issue would help.

**Comment by jmikrut** [comment]
@ljharb — I can give some more info here.

@samicodesit is hitting the same issue that I am. Although `arrayLimit` is an option for `qs.parse`, it is _**not**_ an option for `qs.stringify`. So, when we attempt to stringify an array with more than 20 items in the array, the function appears to use `arrayFormat: 'indices'` instead of the desired ``arrayFormat: 'brackets'`.

We just need an option to specify an `arrayLimit` in the `stringify` method. 

Also, THANK YOU for this package. We are heavily using `qs` in [Payload CMS](https://payloadcms.com) and have just run into this same need.

**Comment by anuragsachdeva28** [comment]
Hey, I am facing the same issue, but in my case I am not using `qs-middleware` on my express server. I think there is something internal parsing happening from `express`. Can anyone help?

**Comment by anuragsachdeva28** [comment]
> 

Hey, I am not using `qs-middleware` and facing the same issue discussed above. Maybe it has something to do with `express` server internal handling. Can you help?

**Comment by alexmachina** [comment]
This should be reopened.
I am currently not able to stringify and parse an array with more than 21 items using `qs`, given that
[qs.parse doesn't handle encoded comma when parsing array with comma option](https://github.com/ljharb/qs/issues/410)
https://github.com/ljharb/qs/issues/410

**Comment by ljharb** [comment]
@alexmachina if you set `arrayLimit`, it should handle it, and the issue you linked is fixed in v6.10.2+.

Please file a new issue if you're still having trouble with the latest version of qs.

**Comment by alexmachina** [comment]
Thanks! I solved it parsing with  `{ arrayFormat: 'brackets'}`
